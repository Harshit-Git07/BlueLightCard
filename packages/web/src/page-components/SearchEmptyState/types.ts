export type SearchEmptyStateProps = {
  onSearchCategoryChange: (categoryId: string, categoryName: string) => void;
};
