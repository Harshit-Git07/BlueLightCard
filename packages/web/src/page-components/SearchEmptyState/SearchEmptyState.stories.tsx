import { Meta, StoryFn } from '@storybook/react';
import { SearchEmptyStatePresenter } from './SearchEmptyState';

const componentMeta: Meta<typeof SearchEmptyStatePresenter> = {
  title: 'Components/SearchEmptyState',
  component: SearchEmptyStatePresenter,
  argTypes: {},
  parameters: {
    controls: {
      exclude: ['categories', 'onSearchCategoryChange'],
    },
  },
};

const mockData = {
  data: {
    response: {
      categories: [
        {
          id: '11',
          name: 'Children and toys',
          __typename: 'CategoryMenu',
        },
        {
          id: '3',
          name: 'Days out',
          __typename: 'CategoryMenu',
        },
        {
          id: '1',
          name: 'Electrical and phones',
          __typename: 'CategoryMenu',
        },
        {
          id: '6',
          name: 'Entertainment',
          __typename: 'CategoryMenu',
        },
        {
          id: '8',
          name: 'Fashion',
          __typename: 'CategoryMenu',
        },
        {
          id: '16',
          name: 'Featured',
          __typename: 'CategoryMenu',
        },
        {
          id: '9',
          name: 'Financial and insurance',
          __typename: 'CategoryMenu',
        },
        {
          id: '15',
          name: 'Food and drink',
          __typename: 'CategoryMenu',
        },
        {
          id: '5',
          name: 'Gifts',
          __typename: 'CategoryMenu',
        },
        {
          id: '4',
          name: 'Health and beauty',
          __typename: 'CategoryMenu',
        },
      ],
    },
  },
};

const DefaultTemplate: StoryFn<typeof SearchEmptyStatePresenter> = (args) => (
  <SearchEmptyStatePresenter {...args}></SearchEmptyStatePresenter>
);

export const Default = DefaultTemplate.bind({});

Default.args = {
  categories: mockData.data.response.categories,
  onSearchCategoryChange: () => {},
};

export default componentMeta;
