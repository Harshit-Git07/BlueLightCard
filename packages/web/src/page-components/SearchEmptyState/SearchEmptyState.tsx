import React, { FC } from 'react';
import { SearchEmptyStateProps } from './types';

type SearchEmptyStatePresenterProps = {};

export const SearchEmptyStatePresenter: FC<SearchEmptyStatePresenterProps> = () => {
  return (
    <div className="w-full">
      <p className="font-typography-title-large text-typography-title-large font-typography-title-large-weight tracking-typography-title-large leading-typography-title-large pb-[25px]">
        No results found
      </p>
      <div className="border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"></div>
    </div>
  );
};

const SearchEmptyState: FC<SearchEmptyStateProps> = () => {
  return <SearchEmptyStatePresenter />;
};

export default SearchEmptyState;
