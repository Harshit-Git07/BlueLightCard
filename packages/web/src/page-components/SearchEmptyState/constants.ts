export const pillGroup = [
  { id: 1, label: 'Beauty and Fragrance' },
  { id: 2, label: 'Food and drink' },
  { id: 3, label: 'Electrical' },
  { id: 4, label: 'Phone' },
  { id: 5, label: 'Fashion' },
  { id: 6, label: 'Children and toys' },
  { id: 7, label: 'Financial and insurance' },
  { id: 8, label: 'Gifts and flowers' },
  { id: 9, label: 'Holiday and travel' },
  { id: 10, label: 'Home' },
  { id: 11, label: 'Jewellery and watches' },
  { id: 12, label: 'Leisure and entertainment' },
  { id: 13, label: 'Motor' },
  { id: 14, label: 'Pets' },
  { id: 15, label: 'Sports and fitness' },
];
