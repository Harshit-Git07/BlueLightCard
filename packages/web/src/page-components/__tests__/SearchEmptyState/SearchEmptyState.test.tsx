import renderer from 'react-test-renderer';
import ReactDom from 'react-dom';
import SearchEmptyState from '../../SearchEmptyState/SearchEmptyState';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime';

jest.mock('react-dom', () => ({
  ...jest.requireActual<typeof ReactDom>('react-dom'),
  preload: jest.fn(),
}));

let mockRouter: Partial<NextRouter>;

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(),
}));

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

describe('SearchDropDown', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <RouterContext.Provider value={mockRouter as NextRouter}>
        <SearchEmptyState />
      </RouterContext.Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
