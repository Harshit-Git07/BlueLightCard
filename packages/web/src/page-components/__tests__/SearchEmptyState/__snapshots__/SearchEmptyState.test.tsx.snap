// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDropDown should match snapshot 1`] = `
<div
  className="w-full"
>
  <p
    className="font-typography-title-large text-typography-title-large font-typography-title-large-weight tracking-typography-title-large leading-typography-title-large pb-[25px]"
  >
    No results found
  </p>
  <div
    className="border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
  />
</div>
`;
